bugsnag-plugin-android-anr/src/main/java/com/bugsnag/android/AnrPlugin.kt

```diff
@@ -16,12 +22,45 @@ internal class AnrPlugin : BugsnagPlugin {
     }
 
     private fun handleAnr(thread: Thread, client: Client) {
+        // generate a full report as soon as possible, then wait for extra process error info
         val errMsg = "Application did not respond to UI input"
         val exc = BugsnagException("ANR", errMsg, thread.stackTrace)
+        val error = Error.Builder(client.config, exc, client.sessionTracker, thread, true)
+            .severity(Severity.ERROR)
+            .severityReasonType(HandledState.REASON_ANR)
+            .build()
+
+        // wait for the ANR dialog to show, reporting once the information is available
```

Comment:
The [info is available](https://android.googlesource.com/platform/frameworks/base/+/refs/heads/master/services/core/java/com/android/server/am/AppErrors.java#1088) before [the dialog shows](https://android.googlesource.com/platform/frameworks/base/+/refs/heads/master/services/core/java/com/android/server/am/AppErrors.java#1093), but just barely. The delay is caused by dumping various stats about the running processes.